include/mbedtls/ccm.h

```diff
@@ -288,6 +293,166 @@ int mbedtls_ccm_star_auth_decrypt( mbedtls_ccm_context *ctx, size_t length,
                       const unsigned char *input, unsigned char *output,
                       const unsigned char *tag, size_t tag_len );
 
+/**
+ * \brief           This function starts a CCM encryption or decryption
+ *                  operation.
+ *
+ * \note            This function is not implemented in Mbed TLS yet.
+ *
+ * \param ctx       The CCM context. This must be initialized.
+ * \param mode      The operation to perform: #MBEDTLS_CCM_ENCRYPT or
+ *                  #MBEDTLS_CCM_DECRYPT or #MBEDTLS_CCM_STAR_ENCRYPT or
+ *                  #MBEDTLS_CCM_STAR_DECRYPT.
+ * \param iv        The initialization vector. This must be a readable buffer
+ *                  of at least \p iv_len Bytes.
+ * \param iv_len    The length of the IV in bytes.
+ * \param total_add_len    The total length of additional data in bytes.
+ * \param total_input_len  The total length of input data to encrypt or decrypt
```

Comment:
I think it would make more sense to call this the plaintext length (stating explicitly that this excludes the additional data, because “plaintext” can be ambiguous), or the length of the encrypted message (it's both the length what gets encrypted, and the length of the result of the encryption excluding the IV and authentication tag).